Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp

```diff
@@ -84,14 +92,346 @@ namespace ScriptAutomation
         {
             auto func = [numSeconds]()
             {
-                auto ScriptAutomationInterface = ScriptAutomationInterface::Get();
-                auto automationComponent = azrtti_cast<ScriptAutomationSystemComponent*>(ScriptAutomationInterface);
+                auto scriptAutomationInterface = ScriptAutomationInterface::Get();
 
-                automationComponent->SetIdleSeconds(numSeconds);
+                scriptAutomationInterface->SetIdleSeconds(numSeconds);
             };
 
             ScriptAutomationInterface::Get()->QueueScriptOperation(AZStd::move(func));
         }
+
+        void ResizeViewport(int width, int height)
+        {
+            auto operation = [width,height]()
+            {
+                if (Utils::SupportsResizeClientAreaOfDefaultWindow())
+                {
+                    Utils::ResizeClientArea(width, height);
+                }
+                else
+                {
+                    AZ_Error("ScriptAutomation", false, "ResizeClientArea() is not supported on this platform");
+                }
+            };
+
+            ScriptAutomationInterface::Get()->QueueScriptOperation(AZStd::move(operation));
+        }
+
+        void CapturePassTimestamp(const AZStd::string& outputFilePath)
+        {
+            auto operation = [outputFilePath]()
+            {
+                auto scriptAutomationInterface = ScriptAutomationInterface::Get();
+
+                scriptAutomationInterface->StartProfilingCapture();
+                scriptAutomationInterface->PauseAutomation();
+
+                AZ::Render::ProfilingCaptureRequestBus::Broadcast(&AZ::Render::ProfilingCaptureRequestBus::Events::CapturePassTimestamp, outputFilePath);
+            };
+
+            ScriptAutomationInterface::Get()->QueueScriptOperation(AZStd::move(operation));
+        }
+        void CaptureCpuFrameTime(const AZStd::string& outputFilePath)
```

Comment:
Each of the Capture functions below should accept an AZStd::string_view or AZ::IO::PathView

Reply:
I opted not to update these because all the functions are exposed to lua and directly pass the argument string to an ebus that is expecting an AZStd::string argument.